### PR TITLE
Add -XLinearTypes extension

### PR DESCRIPTION
**Pull request by monoidal**
The flag needs documentation (to be done separately).
I've noticed several of the tests that were marked as `expect_broken` now work (this isn't related to the branch); haven't changed them yet.

### CODE REVIEW

File: testsuite/driver/runtests.py [hunk 1]
```diff
@@ -81,7 +81,8 @@ def signal_handler(signal, frame):
     config.only = args.only
     config.run_only_some_tests = True
 
-with open("../skip-tests.txt", 'r') as f:
+from os.path import dirname, abspath
+with open(os.path.join(dirname(dirname(abspath(__file__))), "skip-tests.txt"), 'r') as f:
```
Comment (aspiwack) [inline testsuite/driver/runtests.py]: Does this matter, because we intended to revert the changes to this file for the merge.

Comment (monoidal) [reply testsuite/driver/runtests.py]: This allows to run `make test` (or `make accept`) in `linear` directory, which I found useful and I think will be used a few more times in the future.
